tests/target/issue-binop-comments/binop-comments-back.rs

```diff
@@ -0,0 +1,301 @@
+        context /* condition 3 expression */ &&
+        !res.ends('+')
+    {
+        a + b
+    }
+}
+// Test 12.2 - Pre-comments break to few lines - "&&" and "||"
+fn main() {
+    if context() /* condition 1 expression */ && bounds.len() == 1 /* condition 2 expression */ ||
+        context /* condition 3 expression */ && !res.ends('+')
+    {
+        a + b
+    }
+}
+
+// Test 13.1 - Post-comments break to few lines - only "&&"
+fn main() {
+    if context() && /* condition 1 expression */
+        bounds.len() == 1 && /* condition 2 expression */
+        context && /* condition 3 expression */
+        !res.ends('+')
+    /* closing comment */
+    {
+        a + b
+    }
+}
+// Test 13.2 - Pre and Post-comments break to few lines - only "&&"
+fn main() {
+    if context() /* pre comment 1 */ && /* post comment 1 */
+        bounds.len() == 1 /* pre comment 2 */ && /* post comment 2 */
+        context /* pre comment 3 */ && /* post comment 3 */
+        !res.ends('+')
+    /* closing comment */
+    {
+        a + b
+    }
+}
+
+// Test 14.1 - Each "&&" starts new line
+fn main() {
+    if context.inside_macro() &&
+        bounds.len() == 1 &&
+        context.snippet(self.span).ends_with('+') & !res.ends_with('+')
+    {
+        a + b
+    }
+}
+// Test 14.2 - "||" starts new line
+fn main() {
+    if context.inside_macro() && bounds.len() == 1 ||
+        context.snippet(self.span).ends_with('+') & !res.ends_with('+')
+    {
+        a + b
+    }
+}
+
+// Test 15.1 - Each "&&" starts new line - with comments
+fn main() {
+    if context.inside_macro() /* cond 1 */ &&
+        bounds.len() == 1 /* cond 2 */ &&
+        context.snippet(self.span).ends_with('+') /* cond 3 */ &&
+        !res.ends_with('+')
+    {
+        a + b
+    }
+}
+// Test 15.2 - "||" starts new line - with comments
+fn main() {
+    if context.inside_macro() /* cond 1 */ && bounds.len() == 1 /* cond 2 */ ||
+        context.snippet(self.span).ends_with('+') /* cond 3 */ && !res.ends_with('+')
+    {
+        a + b
+    }
+}
+
+// Test 16 - Each "." and some "&&"/"||" starts new line
+fn main() {
+    if a {
+        if b {
+            if c {
+                if d {
+                    if e {
+                        if item_kind
+                            .is_same_item_kind(&***ppi, self.file_mod_map)
+                            .context
+                            .inside_macro() &&
+                            bounds.len() == 1 ||
+                            context.snippet(self.span).ends_with('+') && !res.ends_with('+')
+                        {
+                            a + b
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// Test 21.1
+fn main() {
+    let x = 2 /* v */
+              /* w */ * /* x */
+        4;
+}
+
+// Test 21.2
+fn main() {
+    let x = 2 /* v */
+              /* w */ * /* x */
+                            /* y */
+        4;
+}
```

Comment:
This one looks pretty off to me, and doesn't seem to comply with the requirements from the style guide for binops (https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/expressions.md#binary-operations)

The operator (`*` in this case) is supposed to be block indented one level, not shifted based on the vertical alignment of the separate comments. I think we should be expecting something more like:

```rust
fn main() {
    let x = 2 /* v */ /* w */  * /* x */ /* y */ 4;
}
```

or if it has to be line breaking, maybe
```rust
fn main() {
    let x = 2 /* v */ /* w */ *
        /* x */ /* y */ 4;
}
```